samtranslator/model/eventsources/push.py

```diff
@@ -115,7 +115,10 @@ def to_cloudformation(self, **kwargs):
 
         events_rule.ScheduleExpression = self.Schedule
         if self.Enabled is not None:
-            events_rule.State = "ENABLED" if self.Enabled else "DISABLED"
+            if isinstance(self.Enabled, bool):
+                events_rule.State = "ENABLED" if self.Enabled else "DISABLED"
+            else:
+                events_rule.State = self.Enabled
```

Comment:
Can `self.Enabled` be an [instrinsic function](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html) at this point?
If so, is there a way to evaluate it?

Reply:
I think that if it's passed directly to CloudFormation, it becomes an intrinsic function, isn't it so?